reference-library/TransientStateLibrary.md

```diff
@@ -0,0 +1,122 @@
+# Transient State Library
+
+The `TransientStateLibrary` is a crucial component of Uniswap V4, providing utility functions for managing transient state in the PoolManager contract. This library handles operations related to reserves, delta counts, and locking state, which are essential for the efficient and secure operation of the Uniswap V4 protocol.
+
+## Key Concepts
+
+### Transient Storage
+
+Uniswap V4 uses transient storage to optimize gas costs and improve efficiency. Transient storage, introduced in EIP-1153, is a way to store data that is only needed for the duration of a transaction, without persisting it to the blockchain's state trie. This is achieved using the `TLOAD` and `TSTORE` opcodes.
+
+Common operations that involve transient state include:
+
+- Checking reserves (`getReserves`)
+- Verifying currency deltas (`currencyDelta`)
+- Syncing currency states (`sync`)
+- Settling currency balances (`settle`)
+
+### Storage Slots
+
+The library defines constant storage slots for different pieces of transient state:
+
+``` solidity
+bytes32 public constant RESERVES_OF_SLOT = 0x1e0745a7db1623981f0b2a5d4232364c00787266eb75ad546f190e6cebe9bd95;
+bytes32 public constant NONZERO_DELTA_COUNT_SLOT = 0x7d4b3164c6e45b97e7d87b7125a44c5828d005af88f9d751cfd78729c5d99a0b;
+bytes32 public constant IS_UNLOCKED_SLOT = 0xc090fc4683624cfc3884e9d8de5eca132f2d0ec062aff75d43c0465d5ceeab23;
+```
+
+These slots are computed using `keccak256` hashes of specific strings, ensuring unique storage locations for each piece of data.
```

Comment:
i think we exclude this from documentation -- not very relevant for developers

Reply:
can we replace this section with an example of `using` syntax for syntactic sugar

```solidity
import {IPoolManager} from "v4-core/src/interfaces/IPoolManager.sol";
import {TransientStateLibrary} from "v4-core/src/libraries/TransientStateLibrary.sol";

contract Example {
    using TransientStateLibrary for IPoolManager;

    function example() external {
        manager.currencyDelta(address(this), currency);
    }
}